aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -34,48 +37,69 @@
+    public DailyEmailAggregationJob(EmailAggregationResources emailAggregationResources, ObjectMapper objectMapper) {
         this.emailAggregationResources = emailAggregationResources;
         this.objectMapper = objectMapper;
-        this.registry = registry;
-
-        processedGauge = registry.gauge("daily.email.aggregation.job.processed", new AtomicInteger(0));
     }
 
     @ActivateRequestContext
     public void processDailyEmail() {
-        LocalDateTime now = LocalDateTime.now(UTC);
-        List<AggregationCommand> aggregationCommands = processAggregateEmails(now);
-
-        List<CompletableFuture<Void>> futures = new ArrayList<>();
-        for (AggregationCommand aggregationCommand : aggregationCommands) {
-            try {
-                final String payload = objectMapper.writeValueAsString(aggregationCommand);
-                futures.add(emitter.send(payload).toCompletableFuture());
-            } catch (JsonProcessingException e) {
-                LOG.warn("Could not transform AggregationCommand to JSON object.", e);
+        CollectorRegistry registry = new CollectorRegistry();
+        Gauge duration = Gauge
+                .build()
+                .name("aggregator_job_duration_seconds")
+                .help("Duration of the aggregator job in seconds.")
+                .register(registry);
+        Gauge.Timer durationTimer = duration.startTimer();
+
+        try {
+            LocalDateTime now = LocalDateTime.now(UTC);
+
+            List<AggregationCommand> aggregationCommands = processAggregateEmails(now, registry);
+            List<CompletableFuture<Void>> futures = new ArrayList<>();
+            for (AggregationCommand aggregationCommand : aggregationCommands) {
+                try {
+                    final String payload = objectMapper.writeValueAsString(aggregationCommand);
+                    futures.add(emitter.send(payload).toCompletableFuture());
+                } catch (JsonProcessingException e) {
+                    LOG.warn("Could not transform AggregationCommand to JSON object.", e);
+                }
+
+                // resolve completable futures so the Quarkus main thread doesn't stop before everything has been sent
+                try {
+                    CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get();
+                } catch (InterruptedException | ExecutionException ie) {
+                    LOG.error("Writing AggregationCommands failed", ie);
+                }
             }
 
-            // resolve completable futures so the Quarkus main thread doesn't stop before everything has been sent
+            emailAggregationResources.updateLastCronJobRun(now);
+
+            Gauge lastSuccess = Gauge
+                    .build()
+                    .name("aggregator_job_last_success")
+                    .help("Last time the aggregator job succeeded.")
+                    .register(registry);
+            lastSuccess.setToCurrentTime();
```

Comment:
Does this automatically use the UTC timezone?

Reply:
![Screenshot from 2021-12-13 08-51-39](https://user-images.githubusercontent.com/8372856/145773142-d31c834f-da88-48cc-8f69-a1e27c00ad57.png)

yes, it does